test/tests/functional/pbs_pbsnodes.py

```diff
@@ -54,6 +55,74 @@ def setUp(self):
         self.svrname = self.server.pbs_server_name
         self.hostA = self.moms.values()[0].shortname
 
+    def common_setUp(self):
+        """
+        Common setUp for tests test_pbsnodes_as_user and test_pbsnodes_as_root
+        """
+        TestFunctional.setUp(self)
+        self.server.manager(MGR_CMD_DELETE, NODE, id="",
+                            sudo=True, expect=True)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.mom.shortname)
+        self.server.expect(NODE, {'state': 'free'})
+
+    def get_newnode_attrs(self, user):
+        """
+        return expected values of attributes on a newly created node
+        """
+        expect_dict = {}
+        host_short_name = self.mom.shortname
+        host_name = self.mom.hostname
+        pbs_version = self.server.pbs_version
+
+        expect_dict['Mom'] = host_name
+        expect_dict['ntype'] = 'PBS'
+        expect_dict[ATTR_NODE_state] = 'free'
+        expect_dict['resources_available.vnode'] = host_short_name
+        expect_dict['resources_available.host'] = host_short_name
+        expect_dict['resv_enable'] = 'True'
+
+        if user == 'root':
+            expect_dict['pbs_version'] = pbs_version
+            expect_dict['Port'] = '15002'
+
+        return expect_dict
+
+    def verify_node_dynamic_val(self, last_state_change_time, available_ncpus,
+                                pcpus, sharing, available_mem):
+        """
+        verifies node dynamic attributes have expected value
+        """
+        sharing_list = ['default_shared', 'default_excl', 'default_exclhost',
+                        'ignore_excl', 'force_excl', 'force_exclhost']
+
+        # verify 'last_state_change_time' matched with datetime
+        last_state_change_time = str(last_state_change_time)
+        if time.strptime(last_state_change_time, "%a %b %d %H:%M:%S %Y"):
+            self.logger.info("last_state_change_time matched with datetime")
+
+        # checking resources_avalable.ncpus and pcpus value are positive value
+        ncpus_val = int(available_ncpus)
+        if ncpus_val >= 0:
```

Comment:
this if needs an else case to fail the test if the value is not negative. Same goes for pcpus as well.